### PR TITLE
Feat/init refactor

### PR DESCRIPTION
**Pull request by dblencowe**
Tested on my Mk2 with the alba-medium voice
Can be tested with 
```
pip uninstall ovos-tts-plugin-piper && \
  pip install git+https://github.com/dblencowe/ovos-tts-plugin-piper@feat/init-refactor
sudo systemctl restart neon-audio
```

### CODE REVIEW

**Comment by goldyfruit** [comment]
Why all the changes in `CHANGELOG.md`?

**Comment by dblencowe** [comment]
> Why all the changes in `CHANGELOG.md`?

Awful Git skills after having a mare with the Pipelines. Sorry!

**Comment by goldyfruit** [comment]
> > Why all the changes in `CHANGELOG.md`?
> 
> Awful Git skills after having a mare with the Pipelines. Sorry!

No worries :+1:

**Review by builderjer — COMMENTED** [review]
This works when downloading from a `tz` file but fails to load when pointed to an `onyx` file.  

```
2024-01-25 06:52:24.805 - audio - ovos_tts_plugin_piper:download_model:335 - INFO - downloading piper model: https://huggingface.co/rhasspy/piper-voices/blob/v1.0.0/en/en_GB/vctk/medium/en_GB-vctk-medium.onnx
2024-01-25 06:52:25.661 - audio - ovos_plugin_manager.tts:create:215 - ERROR - The TTS plugin "ovos-tts-plugin-piper" could not be loaded.
Available modules: ovos-tts-plugin-piper,ovos-tts-plugin-dummy,ovos-tts-plugin-server,ovos-tts-plugin-mimic3-server,ovos-tts-plugin-mimic
Traceback (most recent call last):
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_plugin_manager/tts.py", line 206, in create
    tts = clazz(lang=None,  # explicitly read lang from config
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 267, in __init__
    self.get_model(voice=self.voice)
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 317, in get_model
    engine = self.load_model_directory(xdg_p)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 355, in load_model_directory
    config_dict = json.load(config_file)
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 293, in load
    return loads(fp.read(),
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2024-01-25 06:52:25.665 - audio - ovos_audio.service:__init__:93 - ERROR - Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/home/ovos/ovos-src/ovos-audio/ovos_audio/service.py", line 90, in __init__
    self._maybe_reload_tts()
  File "/home/ovos/ovos-src/ovos-audio/ovos_audio/service.py", line 318, in _maybe_reload_tts
    self.tts = TTSFactory.create(config)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/ovos-src/ovos-audio/ovos_audio/tts.py", line 9, in create
    return OVOSTTSFactory.create(config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_plugin_manager/tts.py", line 206, in create
    tts = clazz(lang=None,  # explicitly read lang from config
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 267, in __init__
    self.get_model(voice=self.voice)
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 317, in get_model
    engine = self.load_model_directory(xdg_p)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 355, in load_model_directory
    config_dict = json.load(config_file)
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 293, in load
    return loads(fp.read(),
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2024-01-25 06:52:25.675 - audio - ovos_audio.service:on_error:44 - ERROR - Audio service failed to launch (Expecting value: line 1 column 1 (char 0)).
```

Tested with multiple voices

**Review by mikejgray — COMMENTED** [review]
I'll try to test tonight

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 ovos-plugin-manager>=0.0.5
-piper-tts==1.2.0
+piper-tts==1.1.0
+onnxruntime==1.16.3
```
Comment (goldyfruit) [inline requirements.txt:3]: onnxruntime is already a dependency of `piper-tts`; https://github.com/rhasspy/piper/blob/master/src/python_run/requirements.txt#L2

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 ovos-plugin-manager>=0.0.5
-piper-tts==1.2.0
+piper-tts==1.1.0
```
Comment (goldyfruit) [inline requirements.txt:2]: Any reason to not use `1.2.0`?

Comment (JarbasAl) [reply requirements.txt:2]: 1.2.0 drops support for python < 3.9

Comment (goldyfruit) [reply requirements.txt:2]: Python 3.9 is old, that should not be a blocker to move on IMO.

File: requirements.txt (hunk 1 above)
Comment (dblencowe) [reply requirements.txt:3]: I thought that should be the case, but installs within the Neon docker container were failing as that dependency wasn't installed

File: requirements.txt (hunk 2 above)
Comment (dblencowe) [reply requirements.txt:2]: If we changed the version constraint to >=1.1.0 would Pip be smart enough to figure out the latest version based on Python version?

**Comment by dblencowe** [comment]
> This works when downloading from a `tz` file but fails to load when pointed to an `onyx` file.
> 
> ```
> 2024-01-25 06:52:24.805 - audio - ovos_tts_plugin_piper:download_model:335 - INFO - downloading piper model: https://huggingface.co/rhasspy/piper-voices/blob/v1.0.0/en/en_GB/vctk/medium/en_GB-vctk-medium.onnx
> 2024-01-25 06:52:25.661 - audio - ovos_plugin_manager.tts:create:215 - ERROR - The TTS plugin "ovos-tts-plugin-piper" could not be loaded.
> Available modules: ovos-tts-plugin-piper,ovos-tts-plugin-dummy,ovos-tts-plugin-server,ovos-tts-plugin-mimic3-server,ovos-tts-plugin-mimic
> Traceback (most recent call last):
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_plugin_manager/tts.py", line 206, in create
>     tts = clazz(lang=None,  # explicitly read lang from config
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 267, in __init__
>     self.get_model(voice=self.voice)
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 317, in get_model
>     engine = self.load_model_directory(xdg_p)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 355, in load_model_directory
>     config_dict = json.load(config_file)
>                   ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/__init__.py", line 293, in load
>     return loads(fp.read(),
>            ^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
>     return _default_decoder.decode(s)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
>     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
>     raise JSONDecodeError("Expecting value", s, err.value) from None
> json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
> 2024-01-25 06:52:25.665 - audio - ovos_audio.service:__init__:93 - ERROR - Expecting value: line 1 column 1 (char 0)
> Traceback (most recent call last):
>   File "/home/ovos/ovos-src/ovos-audio/ovos_audio/service.py", line 90, in __init__
>     self._maybe_reload_tts()
>   File "/home/ovos/ovos-src/ovos-audio/ovos_audio/service.py", line 318, in _maybe_reload_tts
>     self.tts = TTSFactory.create(config)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/ovos-src/ovos-audio/ovos_audio/tts.py", line 9, in create
>     return OVOSTTSFactory.create(config)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_plugin_manager/tts.py", line 206, in create
>     tts = clazz(lang=None,  # explicitly read lang from config
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 267, in __init__
>     self.get_model(voice=self.voice)
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 317, in get_model
>     engine = self.load_model_directory(xdg_p)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/ovos/venv-ovos/lib/python3.11/site-packages/ovos_tts_plugin_piper/__init__.py", line 355, in load_model_directory
>     config_dict = json.load(config_file)
>                   ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/__init__.py", line 293, in load
>     return loads(fp.read(),
>            ^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
>     return _default_decoder.decode(s)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
>     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
>     raise JSONDecodeError("Expecting value", s, err.value) from None
> json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
> 2024-01-25 06:52:25.675 - audio - ovos_audio.service:on_error:44 - ERROR - Audio service failed to launch (Expecting value: line 1 column 1 (char 0)).
> ```
> 
> Tested with multiple voices

I'll look at this; I'm using alba-medium (lookup URL from the list) which has downloaded and deployed fine.

File: requirements.txt (hunk 2 above)
Comment (goldyfruit) [reply requirements.txt]: It should.

```
$ pip install "piper-tts>=1.1.0"
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting piper-tts>=1.1.0
  Using cached piper_tts-1.2.0-py3-none-any.whl (29 kB)
Requirement already satisfied: piper-phonemize~=1.1.0 in ./.venvs/ovos/lib/python3.11/site-packages (from piper-tts>=1.1.0) (1.1.0)
Requirement already satisfied: onnxruntime<2,>=1.11.0 in ./.venvs/ovos/lib/python3.11/site-packages (from piper-tts>=1.1.0) (1.16.3)
Requirement already satisfied: coloredlogs in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (15.0.1)
Requirement already satisfied: flatbuffers in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (20181003210633)
Requirement already satisfied: numpy>=1.24.2 in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (1.26.3)
Requirement already satisfied: packaging in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (23.2)
Requirement already satisfied: protobuf in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (5.26.0rc1)
Requirement already satisfied: sympy in ./.venvs/ovos/lib/python3.11/site-packages (from onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (1.12)
Requirement already satisfied: humanfriendly>=9.1 in ./.venvs/ovos/lib/python3.11/site-packages (from coloredlogs->onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (10.0)
Requirement already satisfied: mpmath>=0.19 in ./.venvs/ovos/lib/python3.11/site-packages (from sympy->onnxruntime<2,>=1.11.0->piper-tts>=1.1.0) (1.3.0)
Installing collected packages: piper-tts
Successfully installed piper-tts-1.2.0
```

**Review by mikejgray — CHANGES_REQUESTED** [review]
This worked fine with alan-low on a machine where I already had the model downloaded, but failed with the same error that @builderjer had when trying to use ryan-high

**Comment by JarbasAl** [comment]
replaced by #33

**Pull request closed without merging**